src/PermissionRegistrar.php

```diff
@@ -137,11 +137,15 @@ public function getPermissions(array $params = []): Collection
 
         $permissions = clone $this->permissions;
 
-        foreach ($params as $attr => $value) {
-            $permissions = $permissions->where($attr, $value);
-        }
+        return collect([$permissions->first(function ($item) use ($params) {
```

Comment:
Thanks for submitting this.

Question: since this `getPermissions()` call is initiated by the `Permission` model (ie: typically via `findByName`), which immediately calls `first()` again on this response, is there yet another opportunity here to improve performance even further? I'd be curious about your benchmark results.

Reply:
Hello,

we can't have any performance boost in `findByName` or in `findById`, because `getPermissions()` always returns one item. So the `first()` is happening almost instant.

Reply:
True